includes/core-changes/codeanalysis/5.0/ca1416-platform-compatibility-analyzer.md

```diff
@@ -0,0 +1,127 @@
+
+  ```csharp
+  public void CreateImage()
+  {
+      Image newImage = Image.FromFile("SampImag.jpg");
+  }
+  ```
+
+#### Version introduced
+
+5.0 RC1
+
+#### Recommended action
+
+Ensure that platform-specific APIs are only called when the code is running on an appropriate platform. You can achieve this at build time using preprocessor directives, or at run time by using the platform guards in the <xref:System.OperatingSystem?displayProperty=nameWithType> class:
+
+- Add a `#if` [preprocessor directive](../../../../docs/csharp/language-reference/preprocessor-directives/preprocessor-if.md) around platform-specific API calls:
+
+  ```csharp
+  #if NET50_WINDOWS
+      Console.Beep(261, 1000);
+  #endif
+  ```
+
+- Check the current operating system using one of the <xref:System.OperatingSystem?displayProperty=nameWithType> platform-checking methods, for example, `System.OperatingSystem.IsWindows()`, before calling a platform-specific API. You can use one of these methods in the condition of an `if` statement:
+
+  ```csharp
+  public void PlayCMajor()
+  {
+      if (OperatingSystem.IsWindows())
+      {
+          Console.Beep(261, 1000);
+      }
+  }
+  ```
+
+  ...or pass its result to <xref:System.Debug.Assert(System.Boolean)?displayProperty=nameWithType>:
+
+  ```csharp
+  public void PlayCMajor()
+  {
+      Debug.Assert(OperatingSystem.IsWindows());
+      Console.Beep(261, 1000);
+  }
+  ```
+
+  Use <xref:System.Diagnostics.Debug.Assert(System.Boolean)?displayProperty=nameWithType> when you want the checks to be trimmed out of release builds.
+
+You can also mark your API as platform-specific, in which case the burden of checking requirements falls on your callers. You can mark specific methods or types or an entire assembly.
+
+```csharp
+[SupportedOSPlatform("windows")]
+public void PlayCMajor ()
+{
+    Console.Beep(261, 1000);
+}
+```
+
+If you don't want to fix all your call sites, you can choose one of the following options to suppress the warning:
+
+- To suppress rule CA1416, you can do so using `#pragma` or the [-nowarn](../../../../docs/csharp/language-reference/compiler-options/nowarn-compiler-option.md) compiler flag, or by [setting the rule's severity](../../../../docs/fundamentals/productivity/configure-code-analysis-rules.md#suppress-violations) to `none` in an .editorconfig file.
+
+  ```csharp
+  public void PlayCMajor ()
+  {
+  #pragma warning disable CA1416
+      Console.Beep(261, 1000);
+  #pragma warning restore CA1416
+  }
+  ```
+
+- To disable code analysis completely, set `EnableNETAnalyzers` to `false` in your project file. For more information, see [EnableNETAnalyzers](../../../../docs/core/project-sdk/msbuild-props.md#enablenetanalyzers).
+
+#### Category
+
+- Code analysis
+- Core .NET libraries
+
+#### Affected APIs
+
+For Windows platform:
+
+- All APIs listed at <https://github.com/dotnet/designs/blob/master/accepted/2020/windows-specific-apis/windows-specific-apis.md>.
```

Comment:
I am planning to list all the affected APIs out individually. LMK if that should be done before being published or if it's okay to come later.